File: apps/actor-ios/ActorClient/CocoaMainThreadProvider.swift
//
//  Copyright (c) 2015 Actor LLC. <https://actor.im>
//

import Foundation

class CocoaMainThreadProvider:NSObject, AMMainThreadProvider {

    func postToMainThreadWithRunnable(runnable: JavaLangRunnable!) {
        dispatch_async(dispatch_get_main_queue(), {
            runnable.run()
        });
    }
    
    func isMainThread() -> Bool {
        return NSThread.currentThread().isMainThread
    }
    
    func isSingleThread() -> Bool {
        return false
    }
}

File: actor-sdk/sdk-core/runtime/runtime-cocoa/src/main/java/im/actor/runtime/cocoa/CocoaMainThreadProvider.java
package im.actor.runtime.cocoa;

import im.actor.runtime.MainThreadRuntime;
import im.actor.runtime.os.OSType;

public class CocoaMainThreadProvider implements MainThreadRuntime {

    @Override
    public native void postToMainThread(Runnable runnable)/*-[
        dispatch_async(dispatch_get_main_queue(), ^{
            // NSLog(@"dispatchOnUi(core) %@", runnable);
            [runnable run];
        });
    ]-*/;

    @Override
    public native boolean isMainThread()/*-[
        return [NSThread currentThread].isMainThread;
    ]-*/;

    @Override
    public boolean isSingleThread() {
        return false;
    }

    @Override
    public OSType getOSType() {
        return OSType.IOS;
    }
}
